### PR TITLE
Add complex math functions wrapping C99 functions

### PR DESCRIPTION
**Pull request by daviditen**
Added/modified functions for complex(64) and complex(128) versions of:

abs - absolute value
arg - phase angle
conjg - conjugate
proj - projection onto Riemann sphere
exp - base-e exponent
log - natural logarithm
sqrt - square root

sin, cos, tan - trig functions
asin, acos, atan - arc trig functions
sinh, cosh, tanh - hyperbolic trig functions
asinh, acosh, atanh - arc hyperbolic trig functions

All of these are calls to extern functions defined in C99.  'conjg' already
existed and is the only one of these that is named differently from its C
counterpart, 'conj'.

Added a test to help make sure I got all of the return types right.

### CODE REVIEW

File: modules/standard/Math.chpl [hunk 1]
```diff
@@ -149,11 +149,31 @@ module Math {
...
+  inline proc arg(z: complex(64)): real(32) {
+    extern proc cargf(z: complex(64)): real(32);
+    return cargf(z);
+  }
+
+  /* Returns the real phase angle of complex argument `z`. */
+  inline proc arg(z: complex(128)): real(64) {
+    extern proc carg(z: complex(128)): real(64);
+    return carg(z);
+  }
```
Comment (mppf) [inline modules/standard/Math.chpl]: Since these two apply only to complex numbers and have a peculiar name (arg), I might rather call them 'carg'.

Comment (mppf) [inline modules/standard/Math.chpl]: Why not combine the two arg methods into one, like with abs? If not' I'd consider no-docing one of them.

File: modules/standard/Math.chpl [hunk 2]
```diff
@@ -276,11 +369,29 @@ module Math {
   }
 
 
-  /* Returns the complex conjugate of the argument `a`.
+  /* Returns the complex conjugate of the argument `z`.
    
-     :rtype: The type of `a`.
+     :rtype: The type of `z`.
```
Comment (mppf) [inline modules/standard/Math.chpl]: I'd probably prefer it said rtype is "A complex number of the same type as `z`"

File: modules/standard/Math.chpl [hunk 3]
```diff
@@ -276,11 +369,29 @@ module Math {
...
+
+  /* Returns the projection of `z` on a Riemann sphere. */
+  inline proc proj(z: complex(?w)): real(w/2) {
+    extern proc cprojf(z: complex(64)): real(32);
+    extern proc cproj(z: complex(128)): real(64);
+    if w == 64 then
+      return cprojf(z);
+    else
+      return cproj(z);
+  }
```
Comment (mppf) [inline modules/standard/Math.chpl]: I'd consider naming this one cproj for similar reasons to arg; it only applies to complex and has a potentially likely to conflict name.

**Comment by mppf** [comment]
Hi @daviditen - I had some feedback, overall looks good. I didn't check the chpldoc output but you should sanity check that it looks OK.

I'd prefer you use 'carg' and 'cproj'. My other comments are more minor.

**Comment by daviditen** [comment]
Thanks @mppf .  The changes you suggested sound fine to me.  I'll make those changes and verify that the chpldoc looks good.

**Pull request merged**